### ISSUE TITLE
finetune llama-7B, loss not converge

### ISSUE THREAD

**Issue by LiuPearl1** [tloen/alpaca-lora#224]
I tried to finetune a llama-7B model, and everything gone well. But i found that the loss was not converge, training log like below:
`data_path: /mnt/home/llm/alpaca-lora-main/alpaca_data_cleaned.json
output_dir: ./lora-alpaca
batch_size: 128
micro_batch_size: 4
num_epochs: 3
learning_rate: 0.0003
cutoff_len: 256
val_set_size: 2000
lora_r: 8
lora_alpha: 16
lora_dropout: 0.05
lora_target_modules: ['q_proj', 'v_proj']
train_on_inputs: True
group_by_length: False
wandb_project: 
wandb_run_name: 
wandb_watch: 
wandb_log_model: 
resume_from_checkpoint: False
prompt template: alpaca

Loading checkpoint shards: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 33/33 [00:12<00:00,  2.57it/s]
Found cached dataset json (/root/.cache/huggingface/datasets/json/default-c12b7e070aff57ff/0.0.0/fe5dd6ea2639a6df622901539cb550cf8797e5a6b2dd7af1cf934bed8e233e6e)
100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:00<00:00, 215.99it/s]
trainable params: 4194304 || all params: 6742609920 || trainable%: 0.06220594176090199
Loading cached split indices for dataset at /root/.cache/huggingface/datasets/json/default-c12b7e070aff57ff/0.0.0/fe5dd6ea2639a6df622901539cb550cf8797e5a6b2dd7af1cf934bed8e233e6e/cache-c0511907a8bf8d64.arrow and /root/.cache/huggingface/datasets/json/default-c12b7e070aff57ff/0.0.0/fe5dd6ea2639a6df622901539cb550cf8797e5a6b2dd7af1cf934bed8e233e6e/cache-d1b5cd01b55e5f41.arrow
{'loss': 24937.2625, 'learning_rate': 0.0, 'epoch': 0.03}                                                                                                                           
{'loss': 73800.0063, 'learning_rate': 0.0, 'epoch': 0.05}                                                                                                                           
{'loss': 4949.7477, 'learning_rate': 0.0, 'epoch': 0.08}                                                                                                                            
{'loss': 3093.5197, 'learning_rate': 0.0, 'epoch': 0.1}                                                                                                                             
{'loss': 132576.85, 'learning_rate': 0.0, 'epoch': 0.13}                                                                                                                            
{'loss': 4069595.2, 'learning_rate': 0.0, 'epoch': 0.15}                                                                                                                            
{'loss': 48776.1188, 'learning_rate': 0.0, 'epoch': 0.18}                                                                                                                           
{'loss': 611423.9, 'learning_rate': 0.0, 'epoch': 0.21}                                                                                                                             
{'loss': 27555.4313, 'learning_rate': 0.0, 'epoch': 0.23}                                                                                                                           
{'loss': 3230405.6, 'learning_rate': 0.0, 'epoch': 0.26}                                                                                                                            
{'loss': 16227.8859, 'learning_rate': 0.0, 'epoch': 0.28}                                                                                                                           
{'loss': 81505.6938, 'learning_rate': 0.0, 'epoch': 0.31}                                                                                                                           
{'loss': 23682.7766, 'learning_rate': 0.0, 'epoch': 0.33}                                                                                                                           
{'loss': 41653.4281, 'learning_rate': 0.0, 'epoch': 0.36}                                                                                                                           
{'loss': 54013.1375, 'learning_rate': 0.0, 'epoch': 0.39}                                                                                                                           
{'loss': 46103.8375, 'learning_rate': 0.0, 'epoch': 0.41}                                                                                                                           
{'loss': 469390.55, 'learning_rate': 1.1999999999999999e-05, 'epoch': 0.44}                                                                                                         
{'loss': 48245.5563, 'learning_rate': 4.2e-05, 'epoch': 0.46}                                                                                                                       
 16%|█████████████████████▊                                                                                                                 | 188/1164 [1:58:12<10:11:24, 37.59s/it`

Anyone knows what's wrong with my operation?

Also, I have 4 V110 GPU, and I think that multi-gpu is not work. How should I use multi-gpu? I want to use data parallel.

**Comment by AngainorDev** [comment]
You clearly have an issue.

As for multi gpu training, use torchrun

for instance, with 2 gpus I use

`OMP_NUM_THREADS=4 WORLD_SIZE=2 CUDA_VISIBLE_DEVICES=0,1 torchrun --nproc_per_node=2 --master_port=1234 finetune.py ` and params

**Comment by LiuPearl1** [comment]
@AngainorDev I got it!! Thank you~~

**Comment by pSGAme** [comment]
I also have this issue with one v100, how can I solve it?

**Comment by monologue1107** [comment]
> 

I also meet this problem. Have you solved it?
